WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerViewModel.kt

```diff
@@ -345,6 +356,23 @@ class MediaPickerViewModel @Inject constructor(
         }
     }
 
+    fun onBrowseForItems() {
+        if (mediaPickerSetup.isBrowser()) {
+            clickIcon(ANDROID_CHOOSE_FILE)
```

Comment:
I think this approach could be simplified now that we have only 1 icon to do all the actions. We can have one enum value (`ANDROID_CHOOSE_FROM_DEVICE`) and we can build the allowed mimetypes from the media picker setup allowed types. We can pass this enum and the set of allowed types on icon click (this means this whole logic would disappear).

Reply:
I tried to refactor this a bit also promoting the `MediaPickerIcon` to a sealed class and introducing a few additional types to try to abstract and separate things a bit (MediaPickerIconType, ChooserContext and MediaPickerAction). Let me know if it's over-engineered (or simply wrong 🙃...)